src/connection.go

```diff
@@ -1,13 +1,21 @@
 package main
 
 import (
+	"errors"
 	"fmt"
 	"time"
 
 	"github.com/gomodule/redigo/redis"
 	"github.com/newrelic/infra-integrations-sdk/log"
 )
 
+var (
+	errorConNetworkFailed = errors.New("redis connection failed, redisURL is empty")
+	errorConSocketFailed  = errors.New("redis connection failed, unixSocket is empty and UseUnixSocket is specified")
```

Comment:
I would prefer to not mention CLI flags in errors, and instead check for emptyiness in `main` directly.